### ISSUE TITLE
Parlatype's volume conttrol is... weird

### ISSUE THREAD

**Issue by ferdnyc** [gkarsay/parlatype#10]
I noticed while testing Parlatype that the app's volume control seemed a little disconcerting. (Translation: I nearly deafened myself with a too-loud test file, and found some things puzzling in the process of trying to quiet it.) The app seems to be doing the volume adjustment on its own, instead of leaving that to PulseAudio as I would've expected. Which wouldn't be that big a deal, except the attenuation curve it applies to the output audio stream seems extremely off-kilter.

For the following "demonstration", I generated 5 minutes of 440Hz test tone (a constant-volume file) in Audacity, saved it as a .WAV, and played it in both Parlatype and VLC.

When both players are playing at full volume, the output is the same. (Green highlights.)

![parlatype-vol-100](https://user-images.githubusercontent.com/538020/27939647-4a1785d4-6294-11e7-8b62-d492a108350d.png)

But note how, when I lower VLC's volume to 80%, the PulseAudio stream volume is reduced accordingly. VLC has its volume control linked to PulseAudio's, and allows it to do the attenuation. With Parlatype, even when I set its volume to 80% the PulseAudio stream volume remains at 100%. (Green highlights.)

![parlatype-vol-80](https://user-images.githubusercontent.com/538020/27939716-a8f3d076-6294-11e7-924f-f639d50820b5.png)

As I said earlier, that wouldn't necessarily be a huge issue, on its own. But notice, in this third screenshot, what happens when I lower the volume on both apps (green highlights). Parlatype set to 20% volume is _**louder**_ than VLC/Pulse set to **50% volume** (red highlights)! It's _barely_ reducing the volume. This makes its volume control largely ineffective, because all of the "effective range" for its attenuation is in the lower 20%-30%.

![parlatype-vol-20](https://user-images.githubusercontent.com/538020/27939834-58313f7e-6295-11e7-88e5-f05cdae790f4.png)

Changing the attenuation parameters of Parlatype's volume control would be a help, but really Parlatype should just leave that to PulseAudio. One reason is so that users can adjust the volume _either_ within the app, or from the system settings, the PulseAudio Manager app, or if (like me) they use one of the Gnome Shell advanced-volume extensions, from the system/status menu. With VLC, if I adjust its volume slider in the system/status menu, the volume meter in the app reflects this change, and vice versa (green highlights). For Parlatype, I _can_ use PulseAudio's volume control to adjust the stream volume, but it's a completely separate level setting and leads to confusing interactions.

![parlatype-volmenu](https://user-images.githubusercontent.com/538020/27939981-027819a8-6296-11e7-851e-5fefe343e147.png)

Note that while I use VLC as my (counter)example here, mostly because having a numerical output volume display right in the interface was convenient for my purposes, its behavior is in no way special. Totem works exactly the same, as does Rhythmbox, and Audacious, and any well-behaved Gnome app, really. It's **not** the way Audacity works, notably, but that's because Audacity (while excellent software) is practically the antithesis of "a well-behaved Gnome app". In fact, it still uses Alsa emulation for audio output, instead of talking directly to PulseAudio.

**Comment by gkarsay** [comment]
Thank you very much for your detailed bug report! I believe it's fixed with 2 commits, f9fe421e6cc1c3446fcec9ec89df4ba9fbdc43dd and 7cc3775431925e6714c613802ea2275d1395ad1a

**Comment by ferdnyc** [comment]
After building and installing the git HEAD... looks like, yes! The app's volume control now sets the PulseAudio stream volume, and Parlatype's volume can be controlled by external tools such as the Sound settings pane in Control Center.

Hmm, one minor niggle: While Parlatype is now sending volume-change signals to Pulse, it doesn't look like it's listening for them coming in the other direction. The volume control in the app doesn't react to external volume changes. If I change the stream volume via Gnome's settings app, the Parlatype volume icon doesn't change, nor does the volume slider that opens when it's clicked. Which means that the next time the app's volume control is moved, the stream volume suddenly _JUMP!s_ to the last level it was set to in the app, then returns to following the movements of that slider.

P.S> I also tested out video loading, which looks promising! Loading and playing a short MP4 clip went off without a hitch. When I tried to open a 1.6GB .mkv file — a half-hour program in 1080p HD — I got a "Loading file..." progress meter that was clearly going to take at least **10 minutes** to reach 100%, while it was chewing up so much CPU that there was a 3-4 second typing lag in Firefox. (Firefox prefers that _it_ be the only thing chewing up all of my CPU.) But video support's definitely shaping up! Nicely done.

**Comment by gkarsay** [comment]
Thank you! Can you reproduce the issue (Parlatype's volume slider doesn't move in sync with external volume sliders)? I testet with GTK+ 3.14 and 3.22 and it worked on both. Although this bug might be not fully resolved I will release Parlatype 1.5.1 today or this weekend.

**Comment by ferdnyc** [comment]
Turns out I **cannot** reproduce. I tried it again now (after a pull & build of the latest 1.5.1-devel sources @ 397c4d5, though _nothing_ changed w.r.t. audio output so that shouldn't be relevant) and everything worked perfectly. If the volume control was open in Parlatype, I could even watch it move in real time to follow the slider in the Sound preferences. No idea what I was doing wrong before, but clearly PEBKAC. Withdrawn!
